riptable/rt_str.py

```diff
@@ -915,5 +876,87 @@ def char(self, position: Union[int, List[int], np.ndarray]):
     nb_char_par = _njit_par(_nb_char)
 
 
+def _populate_wrappers(cls):
+    """
+    Decorator for the CatString class which populates the string methods and sets the defaults for
+    filtered value fills.
+    """
+    functions = dict(inspect.getmembers(FAString, inspect.isfunction))
+    properties = dict(inspect.getmembers(FAString, lambda o: isinstance(o, (cached_property, property))))
+    for name, filter_value_fill in [
+        ('upper', None),
+        ('lower', None),
+        ('reverse', None),
+        ('removetrailing', None),
+        ('strlen', INVALID_POINTER_32),

```

Comment:
It's not quite right to use ``INVALID_POINTER_32`` here. The "invalid"/NA value depends on the dtype of the array. A better way to do this might be to put some kind of defined sentinel/marker value here (vs. ``INVALID_POINTER_32``); then in the ``_convert_fastring_output()`` function below, you can detect if the provided fill value is the sentinel, and if so you can get the correct invalid by using the ``.inv`` property of the ``FastArray``.

For example:
```python
class _InvalidFillValueMarker:
    pass

def _populate_wrappers(cls):
    for name, filter_value_fill in [
        # ...
        ('strlen', _InvalidFillValueMarker),
        # ...
    ]:

# ...

def _convert_fastring_output(...):
    if ...:
        # ...
    else:
         if filter_fill_value is _InvalidFillValueMarker:
             filter_fill_value = out.inv
         return where(...)
```